torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -401,6 +402,47 @@ std::shared_ptr<SugaredValue> SugaredDict::attr(
   TORCH_INTERNAL_ASSERT(false);
 }
 
+std::shared_ptr<SugaredEnumClass> SugaredEnumClass::create(
```

Comment:
I see, it is a bit heavy-handed for sure.

We can do two things:
1) don't initialize any values at all, but cache name->value's object mapping
2) create whole name->value mapping upfront, but cache entire SugaredEnumClass object so that next time we can just look up. The cache should always be valid inside same compilation unit because of immutability. Let me know what you think.

Reply:
I think we can just compile the type, bc the type stores the name->value mapping, and use that

Reply:
Type doesn't really store name->value mapping, which I find kind of unnecessary now, contrary to what we had discussed before. Or do you mean something else?